spec/index.bs

```diff
@@ -4835,56 +4837,115 @@ dictionary GPUComputePassDescriptor : GPUObjectDescriptorBase {
 <dl dfn-type=method dfn-for=GPUComputePassEncoder>
     : <dfn>setPipeline(pipeline)</dfn>
     ::
+        Sets the current {{GPUComputePipeline}}.
 
         <div algorithm="GPUComputePassEncoder.setPipeline">
             **Called on:** {{GPUComputePassEncoder}} this.
 
             **Arguments:**
             <pre class=argumentdef for="GPUComputePassEncoder/setPipeline(pipeline)">
-                pipeline:
+                |pipeline|: The compute pipeline to use for subsequent dispatch commands.
             </pre>
 
-            **Returns:** void
+            **Returns:** `void`
 
-            Issue: Describe {{GPUComputePassEncoder/setPipeline()}} algorithm steps.
+            Issue the following steps on the [=Device timeline=] of |this|.{{GPUObjectBase/[[device]]}}:
+            <div class=device-timeline>
+                1. If any of the following conditions are unsatisfied, make |this| [=invalid=] and stop.
+                    <div class=validusage>
+                        - |pipeline| is [$valid to use with$] |this|.
+                    </div>
+                1. Set |this|.{{GPUComputePassEncoder/[[pipeline]]}} to be |pipeline|.
+            </div>
         </div>
 
     : <dfn>dispatch(x, y, z)</dfn>
     ::
+        Dispatch work to be performed with the current {{GPUComputePipeline}}.
 
         <div algorithm="GPUComputePassEncoder.dispatch">
             **Called on:** {{GPUComputePassEncoder}} this.
 
             **Arguments:**
             <pre class=argumentdef for="GPUComputePassEncoder/dispatch(x, y, z)">
-                x:
-                y:
-                z:
+                x: Number of thread groups to dispatch in the X dimension.
```

Comment:
nit: `x` isn't technically a number of thread groups, but `x*y*z` is. So maybe something like "X dimension of the grid of thread groups to dispatch."?

Reply:
SGTM. I had a tough time figuring out what this was supposed to represent from the native specs anyway.

 - **Vulkan**: "groupCountX is the number of local workgroups to dispatch in the X dimension."
 - **D3D12**: "The number of groups dispatched in the x direction."
 - **Metal**: "The number of threadgroups in the grid, in each dimension."